### PR TITLE
Issue1089 external convection

### LINKED ISSUE #1089: External surface convection modelling
The correlation applied by IDEAS.BoundaryConditions.Interfaces.PartialSimInfoManager produces substantially higher convection coefficients for exterior surfaces than the default models used by ESP-r and EnergyPlus.  The correlation is also insensitive to wind direction.  As a consequence each exterior surface of the building is assigned the same convection coefficient for a given time-step.  This situation is illustrated in the figure below.
March_6_Hcext

### PR DESCRIPTION
**Pull request by Mathadon**
replaces #1090 
closes #1089

### CODE REVIEW

**Comment by Mathadon** [comment]
@damienpicard this is ready for review

File: IDEAS/Buildings/Components/Interfaces/SolBus.mo [hunk 1]
```diff
@@ -37,6 +38,12 @@ Connector that contains all solar irridiation information for one inclination an
 </html>", revisions="<html>
 <ul>
 <li>
+November 28, 2019 by Ian Beausoleil-Morrison:<br/>
+Add RealConnector for coefficient for forced convection at exterior surface.
+See <a href=\"https://github.com/open-ideas/IDEAS/issues/1089\">
+#1089</a>
+<\\li>
```
Comment (Mathadon) [inline IDEAS/Buildings/Components/Interfaces/SolBus.mo]:
```suggestion
<\li>
```

File: IDEAS/Buildings/Components/Interfaces/WeaBus.mo [hunk 2]
```diff
@@ -57,6 +59,12 @@ Connector that contains all environment information for many inclinations and ti
    revisions="<html>
 <ul>
 <li>
+November 28, 2019, by Ian Beausoleil-Morrison:<br/>
+Removed hConExt from weather bus because it is not calculated in ExtConvForcedCoeff.mo and ExteriorConvection.mo.<br/>
+Added wind speed and direction to weather bus.
+See <a href=\"https://github.com/open-ideas/IDEAS/issues/1089\">
+#1089</a>
+<li>
```
Comment (Mathadon) [inline IDEAS/Buildings/Components/Interfaces/WeaBus.mo]:
```suggestion
</li>
<li>
```

File: IDEAS/Buildings/Components/OuterWall.mo [hunk 3]
```diff
@@ -243,6 +249,11 @@ The correct shading parameter values should then be passed through the redeclara
 </html>", revisions="<html>
 <ul>
 <li>
+November 28, 2019, by Ian Beausoleil-Morrison:<br/>
+<code>inc</code> and <code>azi</code> of surface now passed as parameters to ExteriorConvection.
+See <a href=\"https://github.com/open-ideas/IDEAS/issues/1089\">
+#1089</a>
+<li>
```
Comment (Mathadon) [inline IDEAS/Buildings/Components/OuterWall.mo]:
```suggestion
<li>
</li>
```

**Comment by damienpicard** [comment]
As you have checked it, I only have review it briefly. I suppose that you looked at the BESTEST and the changes are OK?
This can be merged

File: IDEAS/BoundaryConditions/Interfaces/PartialSimInfoManager.mo [hunk 4]
```diff
@@ -481,12 +493,19 @@ equation
...
-    Diagram(coordinateSystem(preserveAspectRatio=false, extent={{-100,-100},{100,
-            120}})),
+    Diagram(coordinateSystem(preserveAspectRatio=false, extent={{-100,-100},{
+            100,120}})),
     Documentation(info="<html>
 </html>", revisions="<html>
 <ul>
 <li>
+November 28, 2019 by Ian Beausoleil-Morrison:<br/>
+Remove calculation of convection coefficient at exterior surfaces as this has been moved to a new model. Also, removed this coefficient from WeaBus as it is not surface-dependent.<br/>
```
Comment (Mathadon) [inline IDEAS/BoundaryConditions/Interfaces/PartialSimInfoManager.mo]:
```suggestion
Remove calculation of convection coefficient at exterior surfaces 
as this has been moved to a new model. 
Also, removed this coefficient from WeaBus as it is not surface-dependent.<br/>
```

File: IDEAS/BoundaryConditions/SolarIrradiation/RadSol.mo [hunk 5]
```diff
@@ -119,6 +141,13 @@ equation
     Documentation(revisions="<html>
 <ul>
 <li>
+November 28, 2019, by Ian Beausoleil-Morrison:<br/>
+Add RealInput's for wind speed and wind direction.<br/>
+Instantiate extConvForcedCoeff.mo and place resulting coefficient for forced convection on SolBus.
+See <a href=\"https://github.com/open-ideas/IDEAS/issues/1089\">
+#1089</a>
+<\\li>
```
Comment (Mathadon) [inline IDEAS/BoundaryConditions/SolarIrradiation/RadSol.mo]:
```suggestion
<\li>
```

File: IDEAS/BoundaryConditions/SolarIrradiation/RadSolData.mo [hunk 6]
```diff
@@ -171,7 +190,7 @@ Solved bug in linearisation
 </li>
 <li>
 January 21, 2018 by Filip Jorissen:<br/>
-Added <code>solTim</code> connection for revised azimuth computations.
+Added <span style=\"font-family: monospace;\">solTim</span> connection for revised azimuth computations.
```
Comment (Mathadon) [inline IDEAS/BoundaryConditions/SolarIrradiation/RadSolData.mo]:
```suggestion
Added <code>solTim</code> connection for revised azimuth computations.
```

File: IDEAS/Buildings/Components/BaseClasses/ConvectiveHeatTransfer/ExteriorConvection.mo [hunk 7]
```diff
@@ -54,25 +115,20 @@ equation
...
-<img alt=\"equation\" src=\"modelica://IDEAS/Images/equations/equation-HvwkeunV.png\"/> range of [0.15,7.5] meter per second 
-<a href=\"IDEAS.Buildings.UsersGuide.References\">[Defraeye 2011]</a>.
- The <img alt=\"equation\" src=\"modelica://IDEAS/Images/equations/equation-HvwkeunV.png\"/>-dependent term denoting the exterior 
-convective heat transfer coefficient <img alt=\"equation\" src=\"modelica://IDEAS/Images/equations/equation-W7Ft8vaa.png\"/> is 
-determined as <img alt=\"equation\" src=\"modelica://IDEAS/Images/equations/equation-aZcbMNkz.png\"/> in order to take into 
-account buoyancy effects at low wind speeds <a href=\"IDEAS.Buildings.UsersGuide.References\">[Jurges 1924]</a>.
-</p>
+          thickness=0.5)}),
+    Documentation(info="<html>
+<p>The exterior convective heat transfer is computed using Newton&apos;s Law of Cooling. The convection coefficient considers the combined effects of natural (buoyancy driven) and forced (wind-driven) flow.</p>
```
Comment (Mathadon) [inline IDEAS/Buildings/Components/BaseClasses/ConvectiveHeatTransfer/ExteriorConvection.mo]:
```suggestion
<p>
The exterior convective heat transfer is computed using 
Newton's Law of Cooling. The convection coefficient considers the combined 
effects of natural (buoyancy driven) and forced (wind-driven) flow.
</p>
```

File: IDEAS/Buildings/Components/BaseClasses/ConvectiveHeatTransfer/ExteriorConvection.mo [hunk 8]
```diff
@@ -54,25 +115,20 @@ equation
...
+<p>The coefficient for forced convection is calculated in ExtConvForcedCoeff.mo, whereas the current model calculates the coefficient for natural convection and combines the two together to determine the coefficient used in Newton&apos;s Law of Cooling.</p>
```
Comment (Mathadon) [inline IDEAS/Buildings/Components/BaseClasses/ConvectiveHeatTransfer/ExteriorConvection.mo]:
```suggestion
<p>
The coefficient for forced convection is calculated in ExtConvForcedCoeff.mo, 
whereas the current model calculates the coefficient for natural 
convection and combines the two together to determine the 
coefficient used in Newton&apos;s Law of Cooling.
</p>
```

File: IDEAS/Buildings/Components/BaseClasses/ConvectiveHeatTransfer/ExteriorConvection.mo [hunk 9]
```diff
@@ -54,25 +115,20 @@ equation
...
+<p>The &quot;TARP&quot; correlation is used to calculate the coefficient for natural convection. The parameters for the correlation at sourced from the EnergyPlus Engineering manual (Equations 3.75 to 3.77, Page 94). Horizontal surfaces are treated as either a ceiling or a floor, and can be either stably stratified or buoyant, depending upon the surface-to-air temperature difference. Any non-horizontal surface is treated as vertical.</p>
```
Comment (Mathadon) [inline IDEAS/Buildings/Components/BaseClasses/ConvectiveHeatTransfer/ExteriorConvection.mo]:
```suggestion
<p>
The "TARP" correlation is used to calculate the coefficient for natural convection. 
The parameters for the correlation at sourced from the EnergyPlus Engineering 
manual (Equations 3.75 to 3.77, Page 94). 
Horizontal surfaces are treated as either a ceiling or a floor, 
and can be either stably stratified or buoyant, depending 
upon the surface-to-air temperature difference. 
Any non-horizontal surface is treated as vertical.
</p>
```

File: IDEAS/Buildings/Components/BaseClasses/ConvectiveHeatTransfer/ExteriorConvection.mo [hunk 10]
```diff
@@ -54,25 +115,20 @@ equation
...
+<p>This implementation includes a &quot;roughness factor&quot; that can be used to augment the combined convection coefficient calculated by the correlations.  This is used in the so-called &quot;DOE-2 model&quot; that is EnergyPlus&apos; default approach.  However, the use of roughness factors other than 1 are discouraged as there appears to be little physical basis for this factor.</p>
```
Comment (Mathadon) [inline IDEAS/Buildings/Components/BaseClasses/ConvectiveHeatTransfer/ExteriorConvection.mo]:
```suggestion
<p>
This implementation includes a "roughness factor" that can be used to 
augment the combined convection coefficient calculated by the correlations.  
This is used in the so-called "DOE-2 model" that is EnergyPlus'; 
default approach.  However, the use of roughness factors other 
than 1 are discouraged as there appears to be little physical basis for this factor.
</p>
```

File: IDEAS/Buildings/Components/BaseClasses/ConvectiveHeatTransfer/ExteriorConvection.mo [hunk 11]
```diff
@@ -54,25 +115,20 @@ equation
...
 </html>", revisions="<html>
 <ul>
 <li>
+November 28, 2019, by Ian Beausoleil-Morrison:<br/>
+Major rewrite.  Convection coefficients now calculated by surface to consider combined natural and forced effects, and wind direction.
```
Comment (Mathadon) [inline IDEAS/Buildings/Components/BaseClasses/ConvectiveHeatTransfer/ExteriorConvection.mo]:
```suggestion
Major rewrite.  Convection coefficients now calculated by surface to consider 
combined natural and forced effects, and wind direction.
```

File: IDEAS/Buildings/Components/BaseClasses/ConvectiveHeatTransfer/ExteriorConvection.mo [hunk 12]
```diff
@@ -54,25 +115,20 @@ equation
...
+See <a href=\"https://github.com/open-ideas/IDEAS/issues/1089\">
+#1089</a>
+<li>
```
Comment (Mathadon) [inline IDEAS/Buildings/Components/BaseClasses/ConvectiveHeatTransfer/ExteriorConvection.mo]:
```suggestion
</li>
<li>
```

File: IDEAS/Buildings/Components/Interfaces/SolBus.mo [hunk 13]
```diff
@@ -37,6 +38,12 @@ Connector that contains all solar irridiation information for one inclination an
 </html>", revisions="<html>
 <ul>
 <li>
+November 28, 2019 by Ian Beausoleil-Morrison:<br/>
+Add RealConnector for coefficient for forced convection at exterior surface.
+See <a href=\"https://github.com/open-ideas/IDEAS/issues/1089\">
+#1089</a>
+<\li>
```
Comment (Mathadon) [inline IDEAS/Buildings/Components/Interfaces/SolBus.mo]:
```suggestion
</li>
```

**Comment by Mathadon** [comment]
Last version: (June 2018)
bestest

New:
bestest

**Pull request merged**